consensus/fork.go

```diff
@@ -4,119 +4,115 @@ import (
 	"github.com/NebulousLabs/Sia/crypto"
 )
 
-// backtrackToBlockchain returns a list of nodes that go from the current node
-// to the first common parent. The common parent will be the final node in the
-// slice.
-func (s *State) backtrackToBlockchain(bn *blockNode) (nodes []*blockNode) {
-	nodes = append(nodes, bn)
+// invalidateNode recursively moves the children of a blockNode to the bad
+// blocks list.
+func (s *State) invalidateNode(node *blockNode) {
```

Comment:
hmm. `applyTransaction` and `applyMaintenance` are also only called in diffs.go, but I don't think they belong there. Maybe move `invalidateNode` to maintenance.go?